src/java.base/unix/native/libjava/io_util_md.c

```diff
@@ -162,8 +162,12 @@ fileDescriptorClose(JNIEnv *env, jobject this)
             dup2(devnull, fd);
             close(devnull);
         }
-    } else if (close(fd) == -1) {
-        JNU_ThrowIOExceptionWithLastError(env, "close failed");
+    } else {
+        int result;
+        RESTARTABLE(close(fd), result);
```

Comment:
close is not restartable (except AIX apparently). This means you have to explicitly handle EINTR, see UnixNativeDispatcher.close0 for an example.

Reply:
I had it that way before creating the PR; will revert.